railties/lib/rails/generators/rails/model/USAGE

```diff
@@ -67,16 +67,20 @@ Available field types:
 
         `rails generate model product 'price:decimal{10,2}'`
 
+    You can set a `null` or `optional` in curly braces to add a `null: true` option in the migration:
+
+        `rails generate model user pseudo:string{optional}`
+
     You can add a `:uniq` or `:index` suffix for unique or standard indexes
     respectively:
 
         `rails generate model user pseudo:string:uniq`
         `rails generate model user pseudo:string:index`
 
-    You can combine any single curly brace option with the index options:
+    You can combine any curly brace option with the index options:
 
-        `rails generate model user username:string{30}:uniq`
-        `rails generate model product supplier:references{polymorphic}:index`
+        `rails generate model user 'username:string{30,optional}:uniq'`
```

Comment:
I'm not sure `optional` makes sense as an option for non-associations. Strings would be optional by default, right? You can probably revert some changes to `#parse_type_and_options` for this reason.